lib/screens/v2/widget_instance/line_map.ex

```diff
@@ -268,27 +268,26 @@ defmodule Screens.V2.WidgetInstance.LineMap do
   def serialize_scheduled_departure(departures, direction_id, stops, _is_terminal?) do
     # Number of departures with predictions (not just schedules) in this direction
     prediction_count =
-      departures
-      |> Stream.reject(fn %Departure{prediction: p} -> is_nil(p) end)
-      |> Stream.reject(fn %Departure{prediction: %Prediction{trip: t}} -> is_nil(t) end)
-      |> Stream.filter(fn %Departure{prediction: %Prediction{trip: %Trip{direction_id: d}}} ->
-        d == direction_id
-      end)
-      |> Enum.count()
+      Enum.count(
+        departures,
+        &match?(%Departure{prediction: %Prediction{trip: %Trip{direction_id: ^direction_id}}}, &1)
+      )
```

Comment:
**note:** As called out in the PR description this isn't really related to this bug. I'm happy to remove this if you'd like.